### ISSUE TITLE
feat(healthcheck): Add helthcheck api for DefaultService

### ISSUE THREAD

**Issue by AlexCzar** [chrisgahlert/gradle-dcompose-plugin#26]
This is important for programmatic configuration of services in build-scripts.
When using
```kotlin
services.register("name") {
}
```
there should be an option to configure custom healthcheck, analogous to what following docker-compose.yml does:

```yaml
  db:
    image: postgres:10
    ports:
      - "5432:5432"
    environment:
      POSTGRES_PASSWORD: tight5
    healthcheck:
      test: ["CMD", "pg_isready"]
      timeout: 5s
      retries: 3
```
Without this it is impossible to fully configure service programmatically.

**Comment by chrisgahlert** [comment]
Currently only the `HEALTHCHECK` instruction of the Dockerfile is used, when `waitForHealthcheck ` is set to `true`:
https://github.com/chrisgahlert/gradle-dcompose-plugin/wiki/02.-List-of-service-properties#runtime-container-properties

We don't have programmatic way of setting it (yet). Therefore the quick workaround would be:
Create a small `src/test/db/Dockerfile` like this:
```Dockerfile
FROM postgres:10
HEALTHCHECK --interval=3s --retries=56 --start-period=12s CMD pg_isready
```

Then you could create the service by building it:

```gradle
dcompose {
  myDb {
    // before:
    // image = 'postgres:10'

    // after:
    baseDir = file('src/test/db') 
    waitForHealthcheck = true

    // the rest as before:
    portBindings = ['3306:3306']
    ...
  }
}
```

I hope this helps for now! :)

I'll see if I find some time to add a programmatic way of doing it directly in the build script.

**Comment by AlexCzar** [comment]
Yes, this is how I'm currently doing it. But I'd love to get rid of `Dockerfile` files.

**Comment by chrisgahlert** [comment]
This feature will be included in the upcoming release. The syntax will look like this:

```gradle
dcompose {
  myDb {
    image = 'postgres:10'
    waitForHealthcheck = true

    // the new properties:
    healthcheckStartPeriod = 1L // 1ms
    healthcheckInterval = 1000L // 1s
    healthcheckTimeout = 1000L // 1s
    healthcheckRetries = 3
    healthcheckTest = ['CMD', 'pg_isready']

    // the rest as before:
    portBindings = ['3306:3306']
    ...
  }
}
```

For more details about the new properties, see the [wiki page](https://github.com/chrisgahlert/gradle-dcompose-plugin/wiki/02.-List-of-service-properties#runtime-container-properties).

**Comment by chrisgahlert** [comment]
This feature will be included in the upcoming release. The syntax will look like this:

```gradle
dcompose {
  myDb {
    image = 'postgres:10'
    waitForHealthcheck = true

    // the new properties:
    healthcheckStartPeriod = 1L // 1ms
    healthcheckInterval = 1000L // 1s
    healthcheckTimeout = 1000L // 1s
    healthcheckRetries = 3
    healthcheckTest = ['CMD', 'pg_isready']

    // the rest as before:
    portBindings = ['3306:3306']
    ...
  }
}
```

For more details about the new properties, see the [wiki page](https://github.com/chrisgahlert/gradle-dcompose-plugin/wiki/02.-List-of-service-properties#runtime-container-properties).

**Comment by chrisgahlert** [comment]
This has just been released in the latest version `0.16.0`. :)
